File: prisma/seed.ts
import { PrismaClient, MoveStatus } from '@prisma/client';
import bcrypt from 'bcryptjs';

const prisma = new PrismaClient();

const statuses: MoveStatus[] = [
  MoveStatus.YES,
  MoveStatus.NO,
  MoveStatus.HIEROGLYPHS,
  MoveStatus.OUT_OF_STOCK,
  MoveStatus.ALREADY_MOVED,
  MoveStatus.NOT_NEEDED
];

async function main() {
  const password = await bcrypt.hash('password123', 10);
  const workerPassword = await bcrypt.hash('worker123', 10);
  const askelPassword = await bcrypt.hash('white13', 10);

  await prisma.user.upsert({
    where: { email: 'admin@example.com' },
    update: {},
    create: {
      email: 'admin@example.com',
      passwordHash: password,
      role: 'admin',
      displayName: 'Админ'
    }
  });

  await prisma.user.upsert({
    where: { email: 'worker@example.com' },
    update: {},
    create: {
      email: 'worker@example.com',
      passwordHash: workerPassword,
      role: 'worker',
      displayName: 'Исполнитель'
    }
  });

  await prisma.user.upsert({
    where: { email: 'askelwhite22@gmail.com' },
    update: {
      displayName: 'askelwhite22'
    },
    create: {
      email: 'askelwhite22@gmail.com',
      passwordHash: askelPassword,
      role: 'admin',
      displayName: 'askelwhite22'
    }
  });

  const suggestions = [
    { titleMatch: 'steam', path: 'Игры > Steam > Ключи', score: 3 },
    { titleMatch: 'origin', path: 'Игры > Origin > Ключи', score: 2 },
    { titleMatch: 'ubisoft', path: 'Игры > Ubisoft > Ключи', score: 2 }
  ];

  await prisma.suggestion.deleteMany();
  await prisma.suggestion.createMany({ data: suggestions });

  await prisma.item.deleteMany();
  const now = new Date();
  for (let i = 1; i <= 8; i++) {
    const moveStatus = statuses[i % statuses.length];
    await prisma.item.create({
      data: {
        productUrl: `https://ggsel.net/catalog/product/demo-${i}`,
        assigneeName: i % 2 === 0 ? 'Исполнитель' : null,
        moveStatus,
        moveStatusSetBy: 'admin@example.com',
        moveStatusSetAt: now,
        finalBreadcrumbs: i % 2 === 0 ? 'Игры > Steam > Ключи' : null,
        breadcrumbsSetBy: i % 2 === 0 ? 'worker@example.com' : null,
        breadcrumbsSetAt: i % 2 === 0 ? now : null,
        priorityRaw: i % 3 === 0 ? 'Средний' : '',
        completedBy: i % 3 === 0 ? 'worker@example.com' : null,
        completedAt: i % 3 === 0 ? now : null,
        isCompleted: i % 3 === 0,
        movedFlagRaw: i % 2 === 0 ? 'Да' : 'Нет',
        comment: i % 2 === 0 ? 'Комментарий' : null
      }
    });
  }

  await prisma.syncSettings.upsert({
    where: { id: 1 },
    update: {},
    create: { id: 1, autoSyncEnabled: false }
  });
}

main()
  .catch((e) => {
    console.error(e);
    process.exit(1);
  })
  .finally(async () => {
    await prisma.$disconnect();
  });

File: backend/sync/management/commands/seed.py
from __future__ import annotations

from random import choice

from django.core.management.base import BaseCommand
from django.utils import timezone

from accounts.models import User
from items.models import Item
from suggestions.models import Suggestion

SAMPLE_STATUSES = ["Да", "Нет", "Иероглифы", "Нет в наличии", "Уже перенесен", "Перенос не нужен"]


class Command(BaseCommand):
    help = "Seed the database with demo data"

    def handle(self, *args, **options):
        self.stdout.write("Creating users...")
        self._create_users()
        self.stdout.write("Creating items...")
        self._create_items()
        self.stdout.write("Creating suggestions...")
        self._create_suggestions()
        self.stdout.write(self.style.SUCCESS("Seed data created"))

    def _create_users(self):
        users = [
            ("admin@example.com", "password123", User.Role.ADMIN, "Admin"),
            ("askelwhite22@gmail.com", "white13", User.Role.ADMIN, "White"),
            ("worker@example.com", "worker123", User.Role.WORKER, "Worker"),
        ]
        for email, password, role, display_name in users:
            user, created = User.objects.get_or_create(email=email, defaults={"role": role, "display_name": display_name})
            if created:
                user.set_password(password)
                user.save()

    def _create_items(self):
        if Item.objects.exists():
            return
        now = timezone.now()
        for idx in range(1, 11):
            status = choice(SAMPLE_STATUSES)
            breadcrumbs = f"Игры > Категория {idx}"
            Item.objects.create(
                product_url=f"https://ggsel.net/catalog/product/{1000+idx}",
                assignee_name="Worker",
                move_status=status,
                move_status_set_by="admin@example.com",
                move_status_set_at=now,
                final_breadcrumbs=breadcrumbs,
                breadcrumbs_set_by="worker@example.com",
                breadcrumbs_set_at=now,
                priority_raw="Средний" if idx % 2 == 0 else "Низкий",
                completed_by="worker@example.com" if idx % 2 == 0 else "",
                completed_at=now if idx % 2 == 0 else None,
                moved_flag_raw="",
                comment=f"Комментарий {idx}",
                is_completed=idx % 2 == 0,
                row_index=idx + 1,
            )

    def _create_suggestions(self):
        if Suggestion.objects.exists():
            return
        Suggestion.objects.create(path="Игры > Steam", score=0.9, source=Suggestion.Source.UPLOADED)
        Suggestion.objects.create(path="Игры > Origin", score=0.7, source=Suggestion.Source.UPLOADED)
